Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -188,24 +214,29 @@ void UPSWorldSubsystem::OnGameStateChanged(ECurrentGameState CurrentGameState)
 // Load game from save file or create a new one (does initial load from data table)
 void UPSWorldSubsystem::LoadGameFromSave()
 {
+	// load from data table
+	const UDataTable* ProgressionDataTable = UPSDataAsset::Get().GetProgressionDataTable();
+	checkf(ProgressionDataTable, TEXT("ERROR: 'ProgressionDataTableInternal' is null"));
```

Comment:
we should never crash the game with `checkf` when there is chance of game designer fault. In this case, designer might forget to set Data Table, so it should display warning instead:
```
const UDataTable* ProgressionDataTable = UPSDataAsset::Get().GetProgressionDataTable();
if (!ensureMsgf(ProgressionDataTable, TEXT("ASSERT: [%i] %hs:\n'ProgressionDataTable' is not set!"), __LINE__, __FUNCTION__))
{
	return;
}

//....
```

Reply:
I agree, module should not crash the game. Replaced most of chekf to use ensureMsgf instead https://github.com/h4rdmol/ProgressionSystem/pull/18/commits/2ce0a3e2e5e24bc6903d56fc64b33b7688dac9b7

Reply:
@h4rdmol GetMeshChecked() should still use checkf as function called "Checked" meaning it wants to crash as fatal error

the rest is fine